outlines/models/tokenizer.py

```diff
@@ -0,0 +1,23 @@
+from typing import List, Tuple, Union
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+class Tokenizer:
```

Comment:
Should this subclass `Protocol`? I'm not sure what the difference is with this implementation.